src/uu/cp/src/platform/linux.rs

```diff
@@ -202,10 +211,11 @@ pub(crate) fn copy_on_write(
             if source_is_fifo {
                 copy_fifo_contents(source, dest).map(|_| ())
             } else {
-                match check_sparse_detection(source) {
-                    Ok(true) => clone(source, dest, CloneFallback::SparseCopy),
-                    Ok(false) => clone(source, dest, CloneFallback::FSCopy),
-                    Err(e) => Err(e),
+                match (check_sparse_detection(source), check_dest_is_fifo(dest)) {
+                    (Ok(true), false) => clone(source, dest, CloneFallback::SparseCopy),
+                    (Ok(true), true) => clone(source, dest, CloneFallback::FSCopy),
+                    (Ok(false), _) => clone(source, dest, CloneFallback::FSCopy),
+                    (Err(e), _) => Err(e),
```

Comment:
This `match` seems like it's a very complicated implementation of "logical and". Could it be that something like this would work?
```suggestion
                if check_sparse_detection(source)? && !check_dest_is_fifo(dest)) {
                    clone(source, dest, CloneFallback::SparseCopy)
                } else {
                    clone(source, dest, CloneFallback::FSCopy)
                }
```
I haven't run this, and maybe I'm overlooking something obvious; for example, I'm not sure if `?` can automagically construct a `CopyResult`. It's just that I see `match`-arms with identical code, and wonder if there's a better way to express the logic.